Mathlib/Algebra/Order/Rearrangement.lean

```diff
@@ -42,6 +42,13 @@ convenience.
 
 The case for `Monotone`/`Antitone` pairs of functions over a `LinearOrder` is not deduced in this
 file because it is easily deducible from the `Monovary` API.
+
+## TODO
+
+We could golf this file by adding a typeclass `ExistsAddOrAddOfLE α` stating that
+`a ≤ b → ∃ c, a + c = b ∨ a = b + c`, prove it implies the binary rearrangement inequality and
+provide the instances `ExistsAddOfLE α → ExistsAddOrAddOfLE α` and
+`ExistsAddOrAddOfLE α → ExistsAddOrAddOfLE αᵒᵈ`.
```

Comment:
@eric-wieser, before you irremediably order me to split many things off this PR, can you tell me whether you are happy with the duplication in this file or if you want me to implement the scheme above.

The core issue is that...